source/shared/core_results.cpp

```diff
@@ -210,21 +230,20 @@ SQLRETURN number_to_string( Number* number_data, _Out_ void* buffer, SQLLEN buff
 
     if ( std::is_same<Char, char16_t>::value )
     {
-        std::basic_string<char16_t> str;
-        char *str_num_ptr = &str_num[0], *str_num_end = &str_num[0] + str_num.size();
 
-        for ( const auto &mb : str_num )
+        std::basic_string<char16_t> str;
+                
+        for (const auto &mb : str_num )
         {
-            char16_t ch16;
-            std::mbstate_t mbs = std::mbstate_t();
-
-            int len = mbrtoc16( &ch16, &mb, str_num_end - str_num_ptr, &mbs );
-            if ( len > 0 || len == -3 )
+            size_t cch = SystemLocale::NextChar( CP_ACP, &mb ) - &mb;
```

Comment:
Where's the third parameter of SystemLocale::NextChar() ? A default? What's its value?